src/rgw/rgw_sal_motr.cc

```diff
@@ -2593,7 +2595,11 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
   obj.meta.encode(bl);
   ldpp_dout(dpp, 20) <<__func__<< ": lid=0x" << std::hex << obj.meta.layout_id
                                                            << dendl;
-  if (info.versioning_enabled()) {
+
+  // info.versioned() returns true for bucket versioning suspended and enabled case.
+  // Call the MotrObject::update_version_entries() function, to update existing object version entries 
+  // in both versioning enabled and suspended bucket.
+  if (info.versioned()) {
```

Comment:
in case of bucket versioning suspended the null version should have is_latest = true.
Is this the behavior with this change?

Reply:
Yes, this behavior is achieved with this change.
In the current code, we are calling `update_version_entries()` only in case of versioning enabled bucket. If we suspend versioning on the same bucket, we can see is-latest flag set true for current (null) version and previous latest version. 
Hence, calling update_version_entries() for suspend case, resolves the issue of is-latest flag.

Reply:
Okay. try to add few test results as part of PR description so that I gives confidence and clarity to the reviewer.

Reply:
Added the test-results. 
done